dali/base/dafdesc.cpp

```diff
@@ -3653,6 +3661,47 @@ class CStoragePlaneInfo : public CInterfaceOf<IStoragePlane>
     virtual const char * querySingleHost() const override { return xml->queryProp("@host"); }   // MORE: Likely to be changed to resolve hosts
     virtual unsigned numDefaultSprayParts() const override { return xml->getPropInt("@defaultSprayParts", 1); }
     virtual bool queryDirPerPart() const override { return xml->getPropBool("@subDirPerFilePart", isContainerized()); } // default to dir. per part in containerized mode
+    virtual bool hasStorageApi() const override { return xml->hasProp("storageapi"); }
+    virtual StorageType getStorageType() const override
+    {
+        const char * storageTypeStr = xml->queryProp("storageapi/@type");
+        if (strcmp(storageTypeStr, "azurefile")==0)
+            return StorageType::StorageTypeAzureFile;
+        else if (strcmp(storageTypeStr, "azureblob")==0)
+            return StorageType::StorageTypeAzureBlob;
+        return StorageType::StorageTypeUnknown;
+    }
+    virtual const char * queryStorageApiAccount() const override
+    {
+        if(xml->hasProp("storageapi/@account"))
+            return xml->queryProp("storageapi/@account");
+        else
+            return getenv("AZURE_ACCOUNT_NAME");
+    }
+    virtual const char * queryStorageContainer() const override
+    {
+        return xml->hasProp("storageapi/@container") ? xml->queryProp("storageapi/@container"):"";
+    }
+    virtual const StringBuffer & querySASToken (StringBuffer & token) const override
+    {
+        const char *encodedToken = xml->queryProp("storageapi/@sastoken");
+        if (!isEmptyString(encodedToken))
+        {
+            JBASE64_Decode(encodedToken, token);
+        }
+        else
+        {
+            const char * accountName = queryStorageApiAccount();
+            StringBuffer secretName;
+            secretName.appendf("azure-%s",accountName);
+            StringBuffer encToken;
+            getSecretValue(encToken, "storage", secretName, "key", false);
+            encToken.trimRight();
+            JBASE64_Decode(encToken.str(), token);
```

Comment:
The tokens are not purely alphanumeric.  Originally, I encoded because it was not being processed correctly by the templates.  However, it not needed for secrets, so I'll remove the code requiring base64 encoding.